### PR TITLE
Feat/adding upload input

### PR DESCRIPTION
**Pull request by santosluiz000**
#### What problem is this solving?

Adding upload input on store-form

#### How to test it?

Access https://luizdev3--b2bstore.myvtex.com/registration and use the feature there.

#### Screenshots or example usage:

image

### CODE REVIEW

**Review by klzns — DISMISSED** [review]
(btw, I commited some changes to your branch, make sure to git pull before working on it)

I think using draft-js might be to heavy on the client

Can we change to just use react-dropzone instead? We did something similar in styleguide's Dropzone:
Code: https://github.com/vtex/styleguide/blob/master/react/components/Dropzone/index.js
Docs: https://styleguide.vtex.com/#/Components/Forms/Dropzone

What's is the workspace to test? Can you include in the PR description?

**Comment by santosluiz000** [comment]
> (btw, I commited some changes to your branch, make sure to git pull before working on it)
> 
> I think using draft-js might be to heavy on the client
> 
> Can we change to just use react-dropzone instead? We did something similar in styleguide's Dropzone:
> Code: https://github.com/vtex/styleguide/blob/master/react/components/Dropzone/index.js
> Docs: https://styleguide.vtex.com/#/Components/Forms/Dropzone
> 
> What's is the workspace to test? Can you include in the PR description?

I will see this, thanks!

Workspace: https://luizdev2--b2bstore.myvtex.com/registration

**Comment by fobarros** [comment]
Pra mim ainda aparece upload

File: docs/README.md [hunk 1]
```diff
@@ -32,6 +32,7 @@ Now, you are able to use all blocks exported by the `store-form` app. Check out
 | `form-input.textarea` | Renders a big text field in the form. |
 | `form-input.text` | Renders a small text field in the form which has few available characters. |
 | `form-field-group` | Renders different form blocks (such as `form-input.radiogroup` and `form-input.text`) according to each schema's sub-properties type. |
+| `form-input.upload` | Renders a upload field in the form. |
```
Comment (guerreirobeatriz) [inline docs/README.md:35]:
```suggestion
| `form-input.upload` | Renders an `Upload` field in the form. |
```

**Review by guerreirobeatriz — CHANGES_REQUESTED** [review]

**Comment by santosluiz000** [comment]
> Pra mim ainda aparece upload

Pls, access this link: https://luizdev3--b2bstore.myvtex.com/registration

File: react/components/Upload/Upload.tsx [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+  },
+  uploadBtn: {
+    defaultMessage: 'Upload document',
+    id: 'admin/pages.admin.rich-text-editor.upload-button',
+  },
+  uploadLabel: {
+    defaultMessage: 'Choose an document to upload',
+    id: 'admin/pages.admin.rich-text-editor.upload-document-label',
+  },
+})
```
Comment (gsasouza) [inline react/components/Upload/Upload.tsx:59]: All these messages are necessary? And please change the ids, they all are related to admin rich-text-editor

File: react/components/Upload/utils.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  contentState: ContentState
+) {
+  return contentBlock.findEntityRanges((character: CharacterMetadata) => {
+    const entityKey = character.getEntity()
+    return (
+      entityKey !== null &&
+      contentState.getEntity(entityKey).getType() === 'LINK'
+    )
+  }, callback)
+}
```
Comment (gsasouza) [inline react/components/Upload/utils.tsx:26]: Is this file necessary? Please remove draft-js and draftjs-md-converter from dependencies

File: react/package.json [hunk 4]
```diff
@@ -7,6 +7,8 @@
   },
   "dependencies": {
     "apollo-client": "^2.6.8",
+    "draft-js": "^0.10.5",
+    "draftjs-md-converter": "^1.2.0",
```
Comment (gsasouza) [inline react/package.json:11]: same here

**Review by gsasouza — CHANGES_REQUESTED** [review]
Looks good, but if you are not using, remove draft-js from dependencies

File: react/package.json (hunk 4 above)
Comment (santosluiz000) [reply react/package.json]: Done!

File: react/components/Upload/utils.tsx (hunk 3 above)
Comment (santosluiz000) [reply react/components/Upload/utils.tsx]: Done!

File: react/components/Upload/Upload.tsx (hunk 2 above)
Comment (santosluiz000) [reply react/components/Upload/Upload.tsx]: Yes, they are. I updated the ids to store-form pattern

**Review by gsasouza — APPROVED** [review]

File: react/components/Upload/Upload.tsx [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+  const [imageUrl, setImageUrl] = React.useState<string | undefined>()
+  const [error, setError] = React.useState<string | null>()
+
+  const { inputType = InputTypes.input, pointer, ...rest } = props
+
+  const onDropImage = async (files: any[]) => {
+    setError(null)
+
+    try {
+      if (files?.[0]) {
```
Comment (igorbrasileiro) [inline react/components/Upload/Upload.tsx:41]: This Component should catch only one file?!

File: react/components/Upload/Upload.tsx [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+  })
+
+  const handleAddImage = () => {
+    setIsOpen(false)
+  }
+
+  return (
+    <div className="vtex-styleguide-9-x-dropdown vtex-dropdown" ref={ref}>
+      <span className="db mb3 w-100 c-on-base t-small">
+        Operating Agreement
```
Comment (igorbrasileiro) [inline react/components/Upload/Upload.tsx:91]: Shouldn't internationalize this message?

File: react/components/Upload/Upload.tsx [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+            <div className="absolute flex justify-center items-center top-0 left-0 h-100 w-100 br2 z-1 bg-black-05">
+              <Spinner />
+            </div>
+          )}
+
+          <div className={`flex flex-column ${isLoading && 'o-20'}`}>
+            <div className="mb4">
+              <Input
+                label={intl.formatMessage(messages.addLabel)}
+                onChange={(e: any) => {
```
Comment (igorbrasileiro) [inline react/components/Upload/Upload.tsx:120]:
```suggestion
                onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
```

File: react/components/Upload/Link.tsx [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+  entityKey: string
+  offsetKey: string
+}
+
+const Link = (props: Props) => {
+  const { contentState, entityKey, children } = props
+  const { url } = contentState.getEntity(entityKey).getData()
+
+  return (
+    <a href={url} className="blue">
```
Comment (igorbrasileiro) [inline react/components/Upload/Link.tsx:18]: I don't have too much context, but why not use the render-runtime `Link` component?

**Review by igorbrasileiro — APPROVED** [review]
I don't have too much context, but LGTM. I left some comments

File: react/components/Upload/Link.tsx [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+interface Props {
+  children: React.ElementType[]
+  contentState: ContentState
+  decoratedText: string
+  dir: string | null
+  entityKey: string
+  offsetKey: string
+}
+
+const Link = (props: Props) => {
```
Comment (icazevedo) [inline react/components/Upload/Link.tsx:13]: I think this component isn't being used anymore. You should remove it

File: react/package.json [hunk 10]
```diff
@@ -15,6 +15,7 @@
   },
   "devDependencies": {
     "@apollo/react-testing": "^3.1.3",
+    "@types/draft-js": "^0.10.44",
```
Comment (icazevedo) [inline react/package.json:18]: I think you won't need that anymore

File: react/typings/draft-js.d.ts [hunk 11]
```diff
@@ -0,0 +1,21 @@
+import 'draft-js'
```
Comment (icazevedo) [inline react/typings/draft-js.d.ts:1]: You should remove this file as well

File: react/typings/draftjs-md-converter.d.ts [hunk 12]
```diff
@@ -0,0 +1 @@
+declare module 'draftjs-md-converter'
```
Comment (icazevedo) [inline react/typings/draftjs-md-converter.d.ts:1]: This one too

**Review by icazevedo — CHANGES_REQUESTED** [review]
I've left some suggestions. You should also consider the ones Igor commented

File: react/typings/draft-js.d.ts (hunk 11 above)
Comment (santosluiz000) [reply react/typings/draft-js.d.ts:1]: Done!

File: react/typings/draftjs-md-converter.d.ts (hunk 12 above)
Comment (santosluiz000) [reply react/typings/draftjs-md-converter.d.ts:1]: Done!

File: react/components/Upload/Upload.tsx (hunk 5 above)
Comment (santosluiz000) [reply react/components/Upload/Upload.tsx:41]: Yes!

File: react/components/Upload/Upload.tsx (hunk 6 above)
Comment (santosluiz000) [reply react/components/Upload/Upload.tsx:91]: Done!

File: react/components/Upload/Link.tsx (hunk 8 above)
Comment (santosluiz000) [reply react/components/Upload/Link.tsx]: This file is unecessary. It was removed

File: react/components/Upload/Upload.tsx (hunk 6 above)
Comment (icazevedo) [reply react/components/Upload/Upload.tsx:91]: @santosluiz000 you've added the definitions but you also need to add `IOMessage` here so it can internationalize correctly.

You should also use `defineMessages` function to declare all i18n strings you use inside react. Take a look [here](https://github.com/vtex-apps/store-form/blob/f42b4e84aa7d02f63f31a55249b173fba5106ee0/react/FormSubmit.tsx).

File: react/components/Upload/Upload.tsx [hunk 13]
```diff
@@ -0,0 +1,163 @@
...
+                  setImageUrl(e.target.value)
+                }}
+                placeholder={intl.formatMessage({
+                  id: 'store/form.add-document.placeholder',
+                })}
+              />
+            </div>
+
+            <Button onClick={handleAddImage} size="small" disabled={!imageUrl}>
+              <FormattedMessage id="store/form.add-button" />
```
Comment (icazevedo) [inline react/components/Upload/Upload.tsx:130]: Change it to `IOMessage` after you define the messages.

File: react/components/Upload/Upload.tsx [hunk 14]
```diff
@@ -0,0 +1,163 @@
...
+            </Button>
+
+            <div className="flex flex-column">
+              <span className="db mb3 w-100 c-on-base t-small">
+                <FormattedMessage id="store/form.upload-document-label" />
```
Comment (icazevedo) [inline react/components/Upload/Upload.tsx:135]: Change it to `IOMessage` after you define the messages.

File: react/components/Upload/Upload.tsx [hunk 15]
```diff
@@ -0,0 +1,163 @@
...
+              </span>
+              <div {...getRootProps()} className="flex flex-column">
+                <input {...getInputProps()} />
+                <Button size="small">
+                  <FormattedMessage id="store/form.upload-button" />
```
Comment (icazevedo) [inline react/components/Upload/Upload.tsx:140]: Change it to `IOMessage` after you define the messages.

**Review by icazevedo — CHANGES_REQUESTED** [review]

File: react/components/Upload/Upload.tsx (hunk 13 above)
Comment (santosluiz000) [reply react/components/Upload/Upload.tsx:130]: To avoid declare `defineMessages` its a good practice call id directly?

Like this: `<IOMessage id="store/form.add-button"` ?

Comment (icazevedo) [reply react/components/Upload/Upload.tsx:130]: I think it's good to use `defineMessages` for performance purposes. That way, the React builder will now exactly where to look and which messages are available.

File: react/components/Upload/Upload.tsx (hunk 15 above)
Comment (santosluiz000) [reply react/components/Upload/Upload.tsx]: Done!

File: react/components/Upload/Upload.tsx (hunk 14 above)
Comment (santosluiz000) [reply react/components/Upload/Upload.tsx]: Done!

File: react/components/Upload/Upload.tsx (hunk 13 above)
Comment (santosluiz000) [reply react/components/Upload/Upload.tsx]: Done!

File: react/components/Upload/Upload.tsx (hunk 6 above)
Comment (icazevedo) [reply react/components/Upload/Upload.tsx:107]: This is still here @santosluiz000, this message will not be internationalized if it stays like this.

**Review by icazevedo — CHANGES_REQUESTED** [review]

**Review by icazevedo — APPROVED** [review]

**Review by guerreirobeatriz — APPROVED** [review]
